src/lib/input/input.ts

```diff
@@ -245,6 +245,10 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
       });
     }
 
+    if (_platform.SAFARI) {
+      _elementRef.nativeElement.classList.add(this.controlType + '-safari');
```

Comment:
Can you set a member variable like `this._isSafari = _platform.SAFARI` then you can do a host binding to set the class `'[class.mat-input-safari]': '_isSafari'`.

Reply:
Great! It is done.